backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -0,0 +1,207 @@
+package org.ovirt.engine.core.bll.scheduling.utils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.ovirt.engine.core.common.businessentities.CpuPinningPolicy;
+import org.ovirt.engine.core.common.businessentities.VDS;
+import org.ovirt.engine.core.common.businessentities.VM;
+import org.ovirt.engine.core.common.businessentities.VdsCpuUnit;
+import org.ovirt.engine.core.common.utils.CpuPinningHelper;
+import org.ovirt.engine.core.compat.Guid;
+import org.ovirt.engine.core.vdsbroker.ResourceManager;
+
+public class VdsCpuUnitPinningHelper {
+    @Inject
+    private ResourceManager resourceManager;
+
+    /**
+     * This function will tell if the host is capable to run a given dedicated CPU policy VM.
+     *
+     * The function first will pin the CPU pending resources to the host, then:
```

Comment:
`will pin the CPU pending resources to the host` - what does it mean? you mean associate/correlate?

Reply:
The current VdsCpuUnits are a clone, we "pin" them, to let the logic run.

Reply:
Or we can consider this an implementation detail and just say that the method takes into account the pending resources.